astropy/modeling/mappings.py

```diff
@@ -55,17 +59,20 @@ def inverse(self):
                 "Mappings such as {0} that drop one or more of their inputs "
                 "are not invertible at this time.".format(self.mapping))
 
-        return self.__class__(mapping)
+        inv = self.__class__(mapping)
+        inv._inputs = self._outputs
+        inv._outputs = self._inputs
+        return inv
 
 
 class Identity(Mapping):
-    def __init__(self, n_inputs):
-        mapping = tuple(range(n_inputs))
+    def __init__(self, n_outputs):
+        mapping = tuple(range(n_outputs))
```

Comment:
> ident = Identity(1) does not really tell me how many inputs I will pass

If that's the case then it should.  `Identity(1)` should accept one and only one input.